src/script/lua_api/l_server.cpp

```diff
@@ -429,24 +429,30 @@ int ModApiServer::l_get_worldpath(lua_State *L)
 	return 1;
 }
 
-// sound_play(spec, parameters)
+// sound_play(spec, parameters, [ephemeral])
 int ModApiServer::l_sound_play(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	SimpleSoundSpec spec;
 	read_soundspec(L, 1, spec);
 	ServerSoundParams params;
 	read_server_sound_params(L, 2, params);
-	s32 handle = getServer(L)->playSound(spec, params);
-	lua_pushinteger(L, handle);
+	bool ephemeral = lua_isboolean(L, 3) && readParam<bool>(L, 3);
```

Comment:
`lua_isboolean` should be avoided as it causes issues with Lua's `and` and `or`.

`a and b` will return b if a is truthy, otherwise nil. `a or b` will return `a` if it is truthy, otherwise `b`.

`a` and `b` may not be booleans, which makes using `lua_isboolean` an issue if a condition is ever used on this argument.

Reply:
Then what should I change this to?
By the way, grep finds 20 `lua_isboolean` uses in src/script.